tools/Common/IceRpc.Telemetry.Internal/README.md

```diff
@@ -0,0 +1,40 @@
+# Telemetry
+
+`IceRpc.Protobuf.Tools` and `IceRpc.Slice.Tools` collect anonymous
+telemetry data about general usage. Participation in this anonymous program is
+optional, and you may opt-out if you do not like to share any information.
+
+This package includes the `IceRpc.Telemetry.Internal` client. This client is a
+IceRPC C# client that sends anonymous telemetry data over a secure connection
+to the IceRPC telemetry server during the compilation of Slice and Protobuf
+files. This data includes:
+
+- The version of IceRPC that is being used.
+- The operating system of the host machine.
+- The processor count of the host machine.
+- The amount of memory that the client has.
+- The source of the telemetry data (either `IceRpc.Slice.Tools or
+ `IceRpc.Protobuf.Tools`).
+- The hashed slice files or protobuf files. This is used to determine if
+the compiled files are known to the IceRPC telemetry server, such as the
+example slice files or protobuf files provided in the icerpc-csharp
+repository.
+- If the build requires recompiling any Slice or Protobuf files,
```

Comment:
Not sure which part you are referencing

> If the build requires recompiling any Slice or Protobuf files

If it's this line then we can actually figure out if we need to generate code (i.e if any of the files were updated) for both slice and protobuf. However, after discussing with @pepone I think we dont need to capture this information since we can infer it via if the hashes changed.